### DISCUSSION TITLE
How can filter by latest date in code results after search?

### DISCUSSION THREAD

**Discussion by johnfelipe** [community/community#84774] [Other Feature Feedback, Questions, & Ideas]
I know how can filter by latest date in repositories, how can do it in code section?

**Comment by megamiii** [comment]
Currently, GitHub's code search functionality does not offer a direct way to filter results by the most recent date. While you can use various qualifiers in your search, such as language, file extension, and repository, these do not sort code results chronologically. To find the latest code updates, you might need to manually check the date of the commits in the repositories that appear in your search results.

**Comment by johnfelipe** [comment]
Then only need to open each result for view date?
That is something you need performance or add in roadmap.
Filter like repos in recent date but now for code search results 

May be is not method in variables in URL!?!!??

**Comment by ShHaWkK** [comment]
What would also be nice is to manually check each result to find the dates of the last updates especially for users who manage large volumes of code ??!

**Comment by mustcodes** [comment]
I found the solution myself: Listing repos with given search term (eg. oauth):

response = https://api.github.com/search/repositories?q=oauth&order=desc
Now sort it as

# Sort based on created_at date 
data = sorted(response.json()['items'], key=lambda x: x['created_at'], reverse = True)

**Reply by johnfelipe** [reply]
Pls upload a video with your solution end-to-end

**Reply by alimertcakar** [reply]
^ For python disabled people: this is client side sorting the data and not what your looking for

There is `created` qualifier but it is not supported in code search. I guess its too expensive for github :P

**Reply by alexanderbkl** [reply]
maybe adding sort=updated?

**Comment by bbouhm** [comment]
they fucking patched it because of people searching for sensitive data

**Reply by ghost** [reply]
I found out this existed, now searching for the best code for my projects is hard as fuck now
All I see is stale useless shit code nowadays

**Reply by alexanderbkl** [reply]
did you find any fix

**Comment by hamzaAmjad5163** [comment]
> I know how can filter by latest date in repositories, how can do it in code section?

```
import requests
from datetime import datetime

url = "https://api.github.com/search/code"

# Your search query (e.g., searching for "example" in code)
query = "example"
headers = {
    "Authorization": "token YOUR_GITHUB_TOKEN",
    "Accept": "application/vnd.github.v3+json"
}
params = {
    "q": query,
    "sort": "indexed",  # Sort by the indexed date (latest first)
    "order": "desc"     # Descending order (latest first)
}
response = requests.get(url, headers=headers, params=params)
if response.status_code == 200:
    results = response.json()["items"]
    for result in results:
        repo_name = result["repository"]["full_name"]
        file_path = result["path"]
        indexed_date = result["indexed_at"]
        formatted_date = datetime.strptime(indexed_date, "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d %H:%M:%S")
        
        print(f"Repository: {repo_name}")
        print(f"File: {file_path}")
        print(f"Last Indexed: {formatted_date}")
        print("-" * 40)
else:
    print(f"Failed to fetch data: {response.status_code}")
    print(response.json())
```

Note: 

- Replace `YOUR GITHUB TOKEN` with your actual Gihtub token. 
- The `indexed_at` field in the response indicates when the code was last indexed by **Github**.
- You can modify the `query` to search for sepcific code snippets or files.

i think this might help you out. change parameters, query, where github api token is optional in this code. make request to gihtub api, check if request was successful, and then if successful print the result according to last indexed date. You may change it also to first indexed date, or as you want. 
@johnfelipe @alexanderbkl @ShHaWkK @alimertcakar @WillfulDeceit  If this might be helpfull to you please like my comment and help me by following my github account 
-> `hamzaAmjad5163`
